.github/workflows/release.yaml

```diff
@@ -14,183 +14,207 @@ jobs:
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
-      - name: Set version
-        run: echo "RELEASE_VERSION=$(echo ${{ github.event.milestone.title }} | cut -c 2-)" >> $GITHUB_ENV
-
-      - name: Upload notes
-        uses: actions/create-release@v1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          tag_name: ${{ env.artifactId }}-${{ env.RELEASE_VERSION }}
-          release_name: Korpyhe ${{ env.RELEASE_VERSION }}
-          body_path: release_file.md
+    - name: Checkout develop
+      uses: actions/checkout@v2
+      with:
+        ref: develop
+        token: ${{ secrets.ADMIN_GITHUB_TOKEN }}
+        fetch-depth: 0
+
+    - name: Set up Github credentials
+      run: |
+        git config user.name Gaffer
+        git config user.email github-actions@github.com
+
+    - name: Setup JDK
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+
+    - name: Merge release into develop
+      run: |
+        git checkout ${{ needs.create-release-tag.outputs.branch_name }}
+        git checkout develop
+        git merge ${{ needs.create-release-tag.outputs.branch_name }} --strategy-option theirs
+
+    - name: Update development branch
+      run: |
+        git checkout develop
+        mvn release:update-versions -B
+        git commit -a -m "prepare for next development iteration"
+        git push
 
-  generate-javadoc:
+  update-github-releases:
     runs-on: ubuntu-latest
     needs:
-      - create-release-tag
+    - create-release-tag
+
     steps:
-      - name: Checkout Master
-        uses: actions/checkout@v2
-        with:
-          ref: master
-          token: ${{ secrets.ADMIN_GITHUB_TOKEN }}
-
-      - name: Set up Github credentials
-        run: |
-          git config user.name Gaffer
-          git config user.email github-actions@github.com
-
-      - name: Setup JDK
-        uses: actions/setup-java@v1
-        with:
-          java-version: 1.8
-
-      - name: Set version
-        run: echo "RELEASE_VERSION=$(echo ${{ github.event.milestone.title }} | cut -c 2-)" >> $GITHUB_ENV
-
-      - name: Upload Javadoc
-        run: |
-          git remote set-branches --add origin gh-pages
-          git pull
-          mvn -q clean install -Pquick -Dskip.jar-with-dependencies=true -Dshaded.jar.phase=true
-          mvn -q javadoc:javadoc -Pquick
-          git checkout gh-pages
-          rm -rf uk
-          mv target/site/apidocs/* .
-          git add .
-          git commit -a -m "Updated javadoc - $RELEASE_VERSION"
-          git push
+    - name: Checkout Master
+      uses: actions/checkout@v2
+      with:
+        ref: master
+
+    - name: Create Release Notes
+      uses: docker://decathlon/release-notes-generator-action:2.1.0
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Set version
+      run: echo "RELEASE_VERSION=$(echo ${{ github.event.milestone.title }} | cut -c 2-)" >> $GITHUB_ENV
```

Comment:
The release version looks to be used in the other jobs below as well. It is generated separately every time. It could instead by set once in the `create-release-tag` job and made available to these subsequent jobs as an output.

Reply:
I guess to me it seemed like the same amount of visual gloop to do the whole janky `::output` thing more than once and if the information is available in the standalone job, I prefer to just get it from there. Less state and more get me this thing.